src/library/scala/util/Using.scala

```
@@ -12,131 +12,221 @@
-  * to the one thrown first. This is the case ''unless'' a later exception is
-  * [[scala.util.control.NonFatal fatal]], and the one preceding it is not. In that case,
-  * the first exception is added as a suppressed exception to the fatal one, and the fatal
-  * one is thrown. If an exception is a
-  * [[scala.util.control.ControlThrowable ControlThrowable]], no exception will be added to
-  * it as a suppressed exception.
+  * There are multiple ways to automatically manage resources with `Using`. If you only need
+  * to manage a single resource, the [[Using.apply `apply`]] method is easiest; it wraps the
+  * resource opening, operation, and resource releasing in a `Try`.
   *
-  * @example
   * {{{
   * val lines: Try[List[String]] = Using(resource1) { r1 =>
   *   r1.lines.toList
   * }
   * }}}
   *
-  * @example
+  * If you need to manage multiple resources, [[Using.Manager$.apply `Using.Manager`]] should
+  * be used. It allows the managing of arbitrarily many resources, whose creation, use, and
+  * release are all wrapped in a `Try`.
+  *
   * {{{
-  * val lines: Try[Seq[String]] = for {
-  *   r1 <- Using(resource1)
-  *   r2 <- Using(resource2)
-  *   r3 <- Using(resource3)
-  *   r4 <- Using(resource4)
-  * } yield {
+  * val lines: Try[Seq[String]] = Using.Manager { use =>
+  *   val r1 = use(resource1)
+  *   val r2 = use(resource2)
+  *   val r3 = use(resource3)
+  *   val r4 = use(resource4)
+  *
   *   // use your resources here
   *   r1.lines ++ r2.lines ++ r3.lines ++ r4.lines
+  * }}}
+  *
+  * If you wish to avoid wrapping management and operations in a `Try`, you can use
+  * [[Using.resource `Using.resource`]], which throws any exceptions that occur.
+  *
+  * {{{
+  * val lines: List[String] = Using.resource(resource1) { r1 =>
+  *   r1.lines.toList
   * }
   * }}}
+  *
+  * ==Suppression Behavior==
+  *
+  * If two exceptions are thrown (e.g. by an operation and closing a resource),
+  * one of them rethrown, and the other is
```

“one of them *is* rethrown”?

---

Thanks! good catch

---

E.g., comma after `e.g.`  Unless you've already pushed it. I still don't understand how discarding an exception is OK, but I will try it out and comment later when I am better informed.